client/lib/numalib/detect_linux.go

```diff
@@ -105,46 +109,50 @@ func (*Sysfs) discoverCores(st *Topology) {
 			st.NodeIDs = idset.From[hw.NodeID]([]hw.NodeID{0})
 			const node = 0
 			const socket = 0
-			max, _ := getNumeric[hw.KHz](cpuMaxFile, core)
-			base, _ := getNumeric[hw.KHz](cpuBaseFile, core)
-			st.insert(node, socket, core, Performance, max, base)
+			cpuMax, _ := getNumeric[hw.KHz](cpuMaxFile, readerFunc, core)
+			base, _ := getNumeric[hw.KHz](cpuBaseFile, readerFunc, core)
+			st.insert(node, socket, core, Performance, cpuMax, base)
 			return nil
 		})
 	default:
 		// We found node data, associate cores to nodes
 		_ = st.NodeIDs.ForEach(func(node hw.NodeID) error {
-			s, err := os.ReadFile(fmt.Sprintf(cpulistFile, node))
+			s, err := readerFunc(fmt.Sprintf(cpulistFile, node))
 			if err != nil {
 				return err
 			}
 
 			cores := idset.Parse[hw.CoreID](string(s))
 			_ = cores.ForEach(func(core hw.CoreID) error {
+				// if we get incorrect core number, stop
+				if int(core) >= len(st.Cores) {
+					return nil
+				}
```

Comment:
This should stop the panic. But does it leave us in an inconsistent state elsewhere?